### ISSUE TITLE
Documentation: Clear indication on the requierement of finalizeOrder per payment method

### ISSUE THREAD

**Issue by fthobe** [bigcommerce/checkout-sdk-js#781]
Customer Support as well as developers seem to be confused of the requirement for the finalizeOrder method on hosted payment gateways. Even on the ideas forum of BC wrong answers are given: 
https://support.bigcommerce.com/s/question/0D51B000056Secs/custom-checkout-sdk-hosted-payment-not-working

It would be really helpful, internally as well as externally, if there would be a simple list / table indicating if finalizeOrder is needed or not. Currently the documentation states:

> Also, for some payment methods, the customer may be asked to enter their payment details on an external website. For these methods, you must finalize the order when the customer is redirected back to the checkout page in order to complete the checkout flow. 

Simply stating in a separate document which payment methods need finalize order seems more logical to me.

**Comment by capsula4** [comment]
At the moment, our recommendation is to simply always call it. If order finalisation is not required, a specific exception will be thrown and you could simple catch that type of exception.

Is there any reason why you think this approach wouldn't be sufficient for you?

**Comment by fthobe** [comment]
**TL;DR**
Then just write that in the documentation :)

Hi @capsula4 , thank you for getting back to me. I understand that, but it is nowhere clearly written, the documentation states: 

> Also, for **some payment methods**, the customer may be asked to enter their payment details on an external website. For these methods, you must finalize the order when the customer is redirected back to the checkout page in order to complete the checkout flow.

This implies that a: 
a) some hosted payment methods require finalize order and others don't
b) it does not state that this method is mandatory or at least best practice
c) it results in comments like the one linked above wrongly stating that Adyen requieres finalize order while the process is entirely handled by payments.bigcommerce.com after returning from the payment page. 

Even in 2nd level / 3rd Level support the BC team seems to be confused about the requirements and is struggling to identify if issues are a result of wrong implementation or not and by default. Bugfixing Adyen with them took 6 months, several tickets, two escalations to 2nd / 3rd Level support, refusals by BC to acknowledge a problem for months. 

If I'd be you people I'd write in the documentation the following: 

> Some payment methods, the customer may be asked to enter their payment details on an external website. For some of these payment methods it's requiered to finalize the order on the checkout. While it is not a technical requirement for all payment methods, you are advised to implement it to assure consistent behavior across all payment platforms and gateways.

**Comment by fthobe** [comment]
In addition I would really add for clarity and bugfixing to avoid the problems we had a simple list!!!!

**Comment by bc-0dp** [comment]
Closing as resolved
